os/drivers/lcd/mipi_lcd.c

```diff
@@ -349,5 +417,17 @@ FAR struct lcd_dev_s *mipi_lcdinitialize(FAR struct mipi_dsi_device *dsi, struct
 	}
 	priv->config->backlight(CONFIG_LCD_MAXPOWER);
 
+#if defined(CONFIG_LCD_SW_ROTATION)
+	uint8_t *mem = (uint8_t *)kmm_malloc((CONFIG_LCD_XRES * CONFIG_LCD_YRES * 2 + 1) * NUM_OF_LCD_BUFFER);	// each pixel is 8bit int
```

Comment:
Hi @sunghan-chang  @namanjain7 }

With the enable CONFIG_LCD_SW_ROTATION, for the loadable_ext_psram config is unable to bootup.

(CONFIG_LCD_XRES * CONFIG_LCD_YRES * 2 + 1) * NUM_OF_LCD_BUFFER)
CONFIG_LCD_XRES=480
CONFIG_LCD_YRES=800
CONFIG_LCD_XRES=2
(480*800*2+1)*2 = 1,536,002

up_allocate_kheap: start = 0x60119600 size = 1468928

It should be no enough heap causing the issue.
Please help check on it.

Thank you

Reply:
Hello Mr ZhenBei
We have tested the commit on loadable_ext_ddr. It's working in that.
I think loadable_ext_psram is for 5th revision board and it has less ram compared to next one. So, heap allocation is failing.
You can disable the rotation
CONFIG_LCD_SW_ROTATION=n
It won't crash then. But st7701sn won't display the screen correctly as LANDSCAPE is not supported in hardware side.

Reply:
Hi @namanjain7 
I understand your provided statements clearly because I tested and debugged them yesterday before I comment at here.

I'm clear on the config (CONFIG_LCD_SW_ROTATION is not set) can be disable to prevent the crashes for loadable_ext_psram config.

But in your commit, the default setting enabled CONFIG_LCD_SW_ROTATION=y for all configurations and was merged into the master branch.

When the issue occurred, it wasted our time as we had to review various configurations, settings, and code changes to pinpoint the cause of the problem.

Hi @sunghan-chang I hope you will take note of this.
This is public code, so please ensure it works for different configurations, including the basic boot-up.